framework/src/actions/MaterialDerivativeTestAction.C

```diff
@@ -0,0 +1,182 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "MaterialDerivativeTestAction.h"
+
+#include "Conversion.h"
+#include "MooseEnum.h"
+#include "FEProblemBase.h"
+
+#include "libmesh/fe.h"
+#include "libmesh/string_to_enum.h"
+
+template <>
+InputParameters
+validParams<MaterialDerivativeTestAction>()
+{
+  InputParameters params = validParams<Action>();
+  params.addParam<std::vector<VariableName>>("args",
+                                             "Variables the tested material property depends on.");
+  params.addRequiredParam<MaterialPropertyName>(
+      "prop_name", "Name of the material property to test the derivatives of.");
+  MooseEnum prop_type_enum("Real RankTwoTensor RankFourTensor");
+  params.addParam<MooseEnum>(
+      "prop_type", prop_type_enum, "Type of the material property to test the derivatives of.");
+  params.addParam<unsigned int>(
+      "derivative_order", 0, "Highest order derivative to test derivatives of.");
+  return params;
+}
+
+MaterialDerivativeTestAction::MaterialDerivativeTestAction(const InputParameters & parameters)
+  : Action(parameters),
+    _args(getParam<std::vector<VariableName>>("args")),
+    _prop_name(getParam<MaterialPropertyName>("prop_name")),
+    _prop_type(getParam<MooseEnum>("prop_type").getEnum<PropTypeEnum>()),
+    _derivative_order(getParam<unsigned int>("derivative_order")),
+    _second(false),
```

Comment:
Add a parameter to set this?

Reply:
No, there will not be a parameter, but in the future I will check the element type in the mesh.